### PR TITLE
Add regex verifier to config for user name #796

### PR DESCRIPTION
**Pull request by andrejonathan**
### Description
Created a central class for all regex validation patterns and moved all existing regex from commands to this class.

Could change it later from a key value pair of regex to a kvp  object that holds the description, valid input message as well as the pattern

### Essential Checks:

* [x] Have you added tests where necessary?
* [ x Have you added metric gathering code where necessary (in `EventHandlers` and `CommandHandlers`)?
* [x] Have you added new commands to the displayable Help list?
* [x] Have you updated any necessary documentation?

image

### CODE REVIEW

**Comment by kbristow** [comment]
This doesn't add the external config for these validation patterns. I.e. we need one validation for internal usage vs a different one for open source usage.

**Comment by Buzzglo** [comment]
Dont really understand why we are doing this, is it worth it?

**Comment by andrejonathan** [comment]
Thanks guys will make changes as per discussion

image

File: config/validation.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "AddConfigServer_gitUri": "^ssh:\\/\\/.*$",
+  "CreateTeam_teamName": ".{1,22}",
+  "LinkExistingTeamSlackChannel_newTeamChannel": "^(?!<#).*",
+  "OnboardMember_domainUsername": "^[a-z0-9\\\\._-]{7,}$",
+  "OnboardMember_email": "[^@]+@[^\\.]+\\..+"
+}
```
Comment (Buzzglo) [inline config/validation.json]: why is the name so tightly coupled to the purpose and implementation. ?
I would have expected URI, teamname, teamchannelname, domainUserName etc

Comment (andrejonathan) [reply config/validation.json]: as discussed should be ok, as is tightly coupled by name but not via code ie the names can change easily, cheers

**Pull request merged**